src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,204 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Car car = null;
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+            if (car != null) {
+                car.setDrivers(getDriversForCar(id));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a car with id: " + id, e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            disconnectDriversAndCar(car.getId());
+            connectDriversWith(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update the car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE taxi_service.cars SET taxi_service.cars.is_deleted = TRUE "
+                + "WHERE taxi_service.cars.id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete a car with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * "
+                + "FROM taxi_service.cars_drivers cd "
+                + "WHERE cd.driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Long> carIdList = new ArrayList<>();
+            while (resultSet.next()) {
+                carIdList.add(resultSet.getObject("car_id", Long.class));
+            }
+            List<Car> carList = new ArrayList<>();
+            for (Long id : carIdList) {
+                Optional<Car> optionalCar = get(id);
+                optionalCar.ifPresent(carList::add);
+            }
+            return carList;
```

Comment:
you already have implemented method in this class that returns you a Car, you can just invoke it and put result to 
car list, you can do something like this:
while (resultSet.next()) {
                carList.add(get(resultSet.getLong(your car index column from query)).orElseThrow());
            }
your car index column from query = 1, if your car id in cars_drivers first